lib/checkPullRequestTemplate.js

```diff
@@ -0,0 +1,184 @@
+const KARMA_LINTER = 'linter/karma';
+
+/**
+ * Checks that the pull request's body contains the explanation.
+ * @param {import('probot').Octokit.PullsGetResponse} pullRequest
+ */
+const checkPRExplanation = (pullRequest) => {
+  // Restrict search to the overview section.
+  const overviewText = '## Overview';
+  const overviewHeadingIndex = pullRequest.body.indexOf(overviewText);
+
+  // Checks that there is an overview.
+  if (overviewHeadingIndex !== -1) {
+    // Get the index of the next heading after the overview.
+    const nextHeadingIndex = pullRequest.body.indexOf(
+      '## ',
+      overviewHeadingIndex + overviewText.length
+    );
+
+    const overviewSection = pullRequest.body.substring(
+      overviewHeadingIndex,
+      nextHeadingIndex
+    );
+
+    let explanation = overviewSection.split('\n').find((line) => {
+      return line.startsWith('2');
+    });
+
+    const explanationText = '[Explain here what your PR does and why]';
+    if (explanation.includes(explanationText)) {
+      // The user has not added the explanation.
+      return (
+        'the body of this PR is missing the required description, ' +
+        'please update the body with a description of what this PR does.'
+      );
+    }
+    return '';
+  } else {
+    // There is no overview section in the body of the PR.
+    return (
+      'the body of this PR is missing the overview section, please ' +
+      'update it to include the overview.'
+    );
+  }
+};
+
+/**
+ * Checks that the pull request's body contains the explanation.
+ * @param {import('probot').Octokit.PullsGetResponse} pullRequest
+ */
+const checkEssentialChecklist = (pullRequest) => {
+  const checklistText = '## Essential Checklist';
+  const checklistHeadingIndex = pullRequest.body.indexOf(checklistText);
+
+  if (checklistHeadingIndex !== -1) {
+    const nextHeadingIndex = pullRequest.body.indexOf(
+      '## ',
+      checklistHeadingIndex + checklistText.length
+    );
+
+    let checklistSection = pullRequest.body.substring(
+      checklistHeadingIndex,
+      nextHeadingIndex
+    );
+    const uncheckedItems = checklistSection.split('\n').filter((list) => {
+      return list.startsWith('- [ ]');
+    });
+
+    let message = '';
+
+    uncheckedItems.forEach((item) => {
+      if (item.toLowerCase().includes(KARMA_LINTER)) {
+        // Karma and linter is unchecked.
+
+        message +=
+          'The karma and linter checklist has not been checked, ' +
+          'please make sure to run the frontend tests and lint tests before ' +
+          'pushing. ';
+      }
+      if (item.toLowerCase().includes('allow edits from maintainers')) {
+        if (!pullRequest.maintainer_can_modify) {
+          message +=
+            'The allow edits from maintainers checklist needs to ' +
+            'be ticked so that maintainers can rerun failed tests. Endeavour ' +
+            'to add this by ticking on the check box. ';
+        }
```

Comment:
> Do we need the swap first char function now?

Yes. Here, if both "edit from maintainers" and "karma/lint" are unchecked, then the second message (edit from maintainers) need to start with uppercase.

Reply:
Can you please add a comment to explain the same?

Reply:
I was able to refactor it so that we don't need the swap char function again.